### PR TITLE
609 no announcements when offline mode

### PR DESCRIPTION
**Pull request by ismaelAbogabal**
# Feature requirements 

1. add pre cache for images so if the user disconnects from the internet after the onboarding he will have all images available on the cache storage 

# Logic 
1. Once the user selects his mosque we will trigger a cache request for our caching system if the images aren't existing on the cache they will be fetched from the server and cached 
2. also when the user enters again the cache check will be triggered again 
3. any error on the image like 404 won't prevent the user from entering the app 

# Technecals 

1. Flutter cached image package has no support for programmatically caching the images so 
2.  I also tried another package called  [Fast cached network image](https://pub.dev/packages/fast_cached_network_image) and it also has no support for our needs 
3. I create an alternative to it called mawaqit network image that has a basic caching functionality out of the box 
4. and also all of us as developers to trigger the caching manually

### CODE REVIEW

**Comment by ibrahim-zehhaf-mawaqit** [comment]
Sorry my bad, I didn't understand you 3rd point in Logic part.
It's ok now

File: lib/src/services/mosque_manager.dart [hunk 1]
```diff
@@ -191,6 +196,25 @@ class MosqueManager extends ChangeNotifier with WeatherMixin, AudioMixin, Mosque
...
+  /// Qr, mosque image, mosque logo, announcement image
+  Future<void> preCacheImages() async {
+    final now = DateTime.now();
+    final images = [
+      mosque?.image,
+      mosque?.logo,
+      mosque?.interiorPicture,
+      mosque?.exteriorPicture,
+      "https://mawaqit.net/bundles/app/prayer-times/img/background/${mosqueConfig?.backgroundMotif ?? 5}.jpg",
+      'https://mawaqit.net/static/images/store-qrcode.png?4.89.2',
```
Comment (ibrahim-zehhaf-mawaqit) [inline lib/src/services/mosque_manager.dart:210]: Put these info in constants

**Comment by ibrahim-zehhaf-mawaqit** [comment]
@ismaelAbogabal conflict

**Comment by ismaelAbogabal** [comment]
@ibrahim-zehhaf-mawaqit Conflict solved

**Comment by ismaelAbogabal** [comment]
@ibrahim-zehhaf-mawaqit Any updates regarding this.

File: lib/src/services/mosque_manager.dart (hunk 1 above)
Comment (ibrahim-zehhaf-mawaqit) [reply lib/src/services/mosque_manager.dart:210]: @ismaelAbogabal 🆙

Comment (ismaelAbogabal) [reply lib/src/services/mosque_manager.dart]: @ibrahim-zehhaf-mawaqit Rebased on main branch and moved the strings into constants

**Comment by ibrahim-zehhaf-mawaqit** [comment]
To rebase on main please

**Comment by ismaelAbogabal** [comment]
@ibrahim-zehhaf-mawaqit The rebase is done please check it out.

**Review by ibrahim-zehhaf-mawaqit — APPROVED** [review]

**Pull request merged**